pkg/ottl/expression.go

```diff
@@ -18,85 +18,102 @@ import (
 	"fmt"
 
 	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
 )
 
-type TransformContext interface {
-	GetItem() interface{}
-	GetInstrumentationScope() pcommon.InstrumentationScope
+type ResourceContext interface {
 	GetResource() pcommon.Resource
 }
 
-type ExprFunc func(ctx TransformContext) interface{}
+type InstrumentationScopeContext interface {
+	GetInstrumentationScope() pcommon.InstrumentationScope
+}
+
+type SpanContext interface {
+	GetSpan() ptrace.Span
+}
+
+type MetricContext interface {
```

Comment:
We also need a DataPointContext so that we can get rid of the `GetItem()` calls and casts in the datapoints context.

Reply:
Good catch to actually suggest me to remove them :))

For the `DataPointContext` part and the `GetItem` there, still cannot do it since it needs to be able to return 4 types of DataPoints. I will look into that later.

Reply:
I removed all of them, since we use the struct directly (I don't need any cast). In the resource/scope I have that interface, but the implementation is already in internal, so no need for this to be public.